include/swift/AST/ProtocolConformance.h

```diff
@@ -562,10 +565,12 @@ class NormalProtocolConformance : public RootProtocolConformance,
   NormalProtocolConformance(Type conformingType, ProtocolDecl *protocol,
                             SourceLoc loc, DeclContext *dc,
                             ProtocolConformanceState state, bool isUnchecked,
-                            bool isPreconcurrency)
+                            bool isPreconcurrency,
+                            SourceLoc preconcurrencyLoc)
       : RootProtocolConformance(ProtocolConformanceKind::Normal,
                                 conformingType),
-        Protocol(protocol), Loc(loc), Context(dc) {
+        Protocol(protocol), Loc(loc), PreconcurrencyLoc(preconcurrencyLoc),
+        Context(dc) {
```

Comment:
Could you please add an assert to make sure that `preconcurrencyLoc` is never valid when `isPreconcurrency` is set to `false`?

Reply:
Yes, good idea!